### PR TITLE
feat(avoidance): enable avoidance cancel

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

Improve & enable avoidance canceling behavior.

Please approve :arrow_down: before this PR.
https://github.com/autowarefoundation/autoware_launch/pull/476

## Tests performed

- [x] PASS TIER IV INTERNAL SCENARIOS

## Effects on system behavior

Support avoidance canceling behavior.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp [hunk 1]
```diff
@@ -139,67 +139,59 @@ bool AvoidanceModule::isExecutionReady() const
 ModuleStatus AvoidanceModule::updateState()
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp:139]: orgnized well!

File: planning/behavior_path_planner/include/behavior_path_planner/utils/avoidance/avoidance_module_data.hpp [hunk 2]
```diff
@@ -93,7 +93,7 @@ struct AvoidanceParameters
   bool use_opposite_lane{true};
 
   // enable update path when if detected objects on planner data is gone.
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/include/behavior_path_planner/utils/avoidance/avoidance_module_data.hpp:95]: should be updated?

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM

File: planning/behavior_path_planner/include/behavior_path_planner/utils/avoidance/avoidance_module_data.hpp (hunk 2 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/avoidance/avoidance_module_data.hpp]: I updated breif description and fix param name in doc.

**Pull request merged**